### PR TITLE
fix: skip empty parquet

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Skip writing empty SST file if source does not contain any data.

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
LGTM

File: src/storage/src/compaction/task.rs [hunk 1]
```diff
@@ -175,20 +176,21 @@ impl CompactionOutput {
...
-        Ok(FileMeta {
-            region_id,
-            file_id: output_file_id,
-            time_range,
-            level: self.output_level,
-            file_size,
-        })
+            .await?
+            .map(
+                |SstInfo {
```
Comment (killme2008) [inline src/storage/src/compaction/task.rs:183]: What about implementing `From<SstInfo>` for `FileMeta`? Then we can simplify the code as:
```rust
.await?
.map(Into::into)
```

File: src/storage/src/flush.rs [hunk 2]
```diff
@@ -202,28 +202,28 @@ impl<S: LogStore> FlushJob<S> {
...
-                Ok(FileMeta {
-                    region_id,
-                    file_id,
-                    time_range,
-                    level: 0,
-                    file_size,
-                })
+                    .await?
+                    .map(
+                        |SstInfo {
```
Comment (killme2008) [inline src/storage/src/flush.rs:209]: The same suggestion as above.

File: src/storage/src/compaction/task.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/storage/src/compaction/task.rs:183]: > What about implementing `From<SstInfo>` for `FileMeta`? Then we can simplify the code as:
> 
> ```rust
> .await?
> .map(Into::into)
> ```

`FileMeta` also needs `region_id`, `file_id` and `level`, which `SstInfo` cannot provide.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**